### ISSUE TITLE
More tests...

### ISSUE THREAD

**Issue by ChrisBarker-NOAA** [pyugrid/pyugrid#22]
We really need more tests.

In particular, a larger selection of sample files to test the reading code against. There are a lot of options and variations, so really hard to know if it's working in the general case. In fact, there is only one file in the `test/files/` dir -- so it's the only one being tested against. And that one doesn't have edges defined (so I have no idea if the code that reads the edges_connectivity array is working...

Also: if anyone is using any part of the API -- that part should have a test, so we know not to break it.

**Comment by ocefpaf** [comment]
@ChrisBarker-NOAA you can assign me to this task.

- [ ] test `edges_connectivity`
- [ ] add a dataset with edges define
- [ ] check the API and implement unittests

**Comment by ChrisBarker-NOAA** [comment]
maybe run coverage, too?

**Comment by ocefpaf** [comment]
> maybe run coverage, too?

Yep.  I updated the list above with that.  It should be fairly simple with `py.test`.

**Comment by ocefpaf** [comment]
OK. I have something ready, but to avoid more merge conflicts can we sort #124, #120, and #119 out first?
